### PR TITLE
[CLOUDTRUST-3748] Improve rights management api - New  version

### PR DESCRIPTION
**Pull request by basbeu**

### CODE REVIEW

File: api/management/swagger-api_management.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 openapi: 3.0.1
 info:
   title: Swagger Cloudtrust Management
-  description: 'Management API for Cloudtrust'
+  description: 'Mangement API for Cloudtrust'
```
Comment (harture) [inline api/management/swagger-api_management.yaml:4]: Added a typo

Comment (harture) [reply api/management/swagger-api_management.yaml:4]: Are you sure this PR is rebase on the last version of master ?

**Review by harture — CHANGES_REQUESTED** [review]

**Review by harture — APPROVED** [review]

File: vendor/modules.txt [hunk 2]
```diff
@@ -36,7 +36,7 @@ github.com/VividCortex/gohistogram
 github.com/certifi/gocertifi
 # github.com/clbanning/x2j v0.0.0-20191024224557-825249438eec
 ## explicit
-# github.com/cloudtrust/common-service/v2 v2.5.2
+# github.com/cloudtrust/common-service/v2 v2.5.3-0.20211209095354-0125f4bc4c77
```
[the remark this answers is not in the document]
Comment (basbeu) [reply vendor/modules.txt:39]: Yes, thank you. It was planned to be changed once I merged the PR in the common-service !

File: pkg/management/component.go [hunk 3]
```diff
@@ -1702,6 +1713,192 @@ func (c *component) UpdateAuthorizations(ctx context.Context, realmName string,
 	return nil
 }
 
+func (c *component) AddAuthorization(ctx context.Context, realmName string, groupID string, authz api.AuthorizationsRepresentation) error {
+	err := c.authChecker.ReloadAuthorizations(ctx)
```
[the remark this answers is not in the document]
Comment (basbeu) [reply pkg/management/component.go:1717]: Done

File: pkg/management/component.go [hunk 4]
```diff
@@ -1702,6 +1713,192 @@ func (c *component) UpdateAuthorizations(ctx context.Context, realmName string,
...
+		}
+	}
+
+	c.reportEvent(ctx, "API_AUTHORIZATIONS_PUT", database.CtEventRealmName, realmName, database.CtEventGroupName, groupName)
+
+	return nil
+}
+
+func (c *component) GetAuthorization(ctx context.Context, realmName string, groupID string, targetRealm string, targetGroupID string, actionReq string) (api.AuthorizationMessage, error) {
+	err := c.authChecker.ReloadAuthorizations(ctx)
```
[the remark this answers is not in the document]
Comment (basbeu) [reply pkg/management/component.go:1796]: Done

File: pkg/management/component.go [hunk 5]
```diff
@@ -1702,6 +1713,192 @@ func (c *component) UpdateAuthorizations(ctx context.Context, realmName string,
...
+			var exists error
+			if scope == security.ScopeGroup {
+				exists = c.authChecker.CheckAuthorizationForGroupsOnTargetGroup(*authz.RealmID, []string{*authz.GroupName}, *authz.Action, *authz.TargetRealmID, *authz.TargetGroupName)
+			} else {
+				exists = c.authChecker.CheckAuthorizationForGroupsOnTargetRealm(*authz.RealmID, []string{*authz.GroupName}, *authz.Action, *authz.TargetRealmID)
+			}
+
+			// If the authorization does not exist yet
+			if exists != nil {
+				// clean children
```
[the remark this answers is not in the document]
Comment (basbeu) [reply pkg/management/component.go:1763]: Done

File: pkg/management/component.go [hunk 6]
```diff
@@ -1702,6 +1713,192 @@ func (c *component) UpdateAuthorizations(ctx context.Context, realmName string,
...
+	}
+
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+	group, err := c.keycloakClient.GetGroup(accessToken, realmName, groupID)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return api.AuthorizationMessage{}, err
+	}
+
+	targetGroupName := &targetGroupID
```
[the remark this answers is not in the document]
Comment (basbeu) [reply pkg/management/component.go:1809]: Done

File: pkg/management/component.go [hunk 7]
```diff
@@ -1702,6 +1713,192 @@ func (c *component) UpdateAuthorizations(ctx context.Context, realmName string,
...
+		TargetGroupName: targetGroupName,
+	}
+
+	err = validateScope(authz)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return api.AuthorizationMessage{}, err
+	}
+
+	scope, _ := getScope(authz)
```
[the remark this answers is not in the document]
Comment (basbeu) [reply pkg/management/component.go:1835]: done

File: pkg/management/component.go [hunk 8]
```diff
@@ -1702,6 +1713,192 @@ func (c *component) UpdateAuthorizations(ctx context.Context, realmName string,
...
+	{
+		tx, err := c.configDBModule.NewTransaction(ctx)
+		if err != nil {
+			c.logger.Warn(ctx, "err", err.Error())
+			return err
+		}
+		defer tx.Close()
+
+		for _, authz := range authorizations {
+			scope, _ := getScope(authz)
```
[the remark this answers is not in the document]
Comment (basbeu) [reply pkg/management/component.go:1752]: done

File: pkg/management/component.go [hunk 9]
```diff
@@ -1702,6 +1713,192 @@ func (c *component) UpdateAuthorizations(ctx context.Context, realmName string,
...
+func (c *component) DeleteAuthorization(ctx context.Context, realmName string, groupID string, targetRealm string, targetGroupID string, actionReq string) error {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+
+	group, err := c.keycloakClient.GetGroup(accessToken, realmName, groupID)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return err
+	}
+
+	targetGroupName := &targetGroupID
```
[the remark this answers is not in the document]
Comment (basbeu) [reply pkg/management/component.go:1858]: done

File: pkg/management/component.go [hunk 10]
```diff
@@ -1702,6 +1713,192 @@ func (c *component) UpdateAuthorizations(ctx context.Context, realmName string,
...
+	if *targetGroupName == "" {
+		targetGroupName = nil
+	} else if *targetGroupName != "*" {
```
[the remark this answers is not in the document]
Comment (basbeu) [reply pkg/management/component.go:1861]: done

File: pkg/management/component.go [hunk 11]
```diff
@@ -1702,6 +1713,192 @@ func (c *component) UpdateAuthorizations(ctx context.Context, realmName string,
...
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+
+	group, err := c.keycloakClient.GetGroup(accessToken, realmName, groupID)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return err
+	}
+
+	targetGroupName := &targetGroupID
+	if *targetGroupName == "" {
```
[the remark this answers is not in the document]
Comment (basbeu) [reply pkg/management/component.go:1859]: done

**Pull request merged**